caringcaribou/tests/mock/mock_ecu_uds.py

```diff
@@ -145,17 +145,25 @@ def handle_read_data_by_identifier(self, data):
         :return: Response to be sent
         """
         service_id = data[0]
+        data_id = data[1:3]
         request = data[2]
 
         if request == self.IDENTIFIER_REQUEST_POSITIVE:
             # Request for positive response
             # TODO Actually read a parameter from memory
-            payload = [self.IDENTIFIER_REQUEST_POSITIVE_RESPONSE]
+            payload = data_id
+            payload.append(self.IDENTIFIER_REQUEST_POSITIVE_RESPONSE)
             response_data = self.create_positive_response(service_id, payload)
         elif request == self.IDENTIFIER_REQUEST_NEGATIVE:
             # Request for negative response - use Conditions Not Correct
             nrc = NegativeResponseCodes.CONDITIONS_NOT_CORRECT
             response_data = self.create_negative_response(service_id, nrc)
+        # special DID that responds with DID + 1
+        elif data_id == [0xff, 0xfe]:
+            print('found special DID')
+            payload = [0xff, 0xff, self.IDENTIFIER_REQUEST_POSITIVE_RESPONSE]
+            print(f'{payload=}')
```

Comment:
While this syntax is both concise and _very_ readable, it requires Python 3.8 or higher. To maintain compatibility with systems running older Python versions, it would be preferable to go for an old fashioned `print(f'payload={payload}')` or similar instead 😅

Reply:
No problem, we can use the older syntax like you suggested. I'm fine with you making the changes :)